packages/experimental/RadioGroup/src/Radio/useRadio.ts

```diff
@@ -0,0 +1,86 @@
+import { RadioProps, RadioState } from './Radio.types';
+import * as React from 'react';
+import { useRadioGroupContext } from '../RadioGroup/radioGroupContext';
+import { useAsPressable, useOnPressWithFocus, useViewCommandFocus } from '@fluentui-react-native/interactive-hooks';
+import { radioSelectActionLabel } from './Radio.styling';
+
+export const useRadio = (props: RadioProps): RadioState => {
+  const defaultComponentRef = React.useRef(null);
+  const {
+    value,
+    disabled,
+    accessibilityLabel,
+    componentRef = defaultComponentRef,
+    accessibilityPositionInSet,
+    accessibilitySetSize,
+    ...rest
+  } = props;
+
+  // Grabs the context information from RadioGroup (currently selected button and client's onChange callback)
+  const info = useRadioGroupContext();
+
+  const buttonRef = useViewCommandFocus(componentRef);
+
+  /* We don't want to call the user's onChange multiple times on the same selection. */
+  const changeSelection = React.useCallback(() => {
+    if (value != info.value) {
+      info.onChange && info.onChange(value);
+      info.updateSelectedButtonRef && componentRef && info.updateSelectedButtonRef(componentRef);
+    }
+  }, [info, value, componentRef]);
+
+  /* We use the componentRef of the currently selected button to maintain the default tabbable
+    element in a RadioGroup. Since the componentRef isn't generated until after initial render,
+    we must update it once here. */
+  React.useEffect(() => {
+    if (value == info.value) {
+      info.updateSelectedButtonRef && componentRef && info.updateSelectedButtonRef(componentRef);
+    }
+  }, [info, value, componentRef]);
+
+  // Ensure focus is placed on button after click
+  const changeSelectionWithFocus = useOnPressWithFocus(componentRef, changeSelection);
+
+  /* RadioButton changes selection when focus is moved between each RadioButton and on a click */
+  const pressable = useAsPressable({
+    ...rest,
+    onPress: changeSelectionWithFocus,
+    onFocus: changeSelection,
+  });
+
+  // Used when creating accessibility properties in mergeSettings below
+  const onAccessibilityAction = React.useCallback(
+    (event: { nativeEvent: { actionName: any } }) => {
+      switch (event.nativeEvent.actionName) {
+        case 'Select':
+          changeSelection();
+          break;
+      }
+    },
+    [changeSelection],
+  );
+
+  const state = {
+    ...pressable.state,
+    selected: info.value === props.value,
+    disabled: disabled || false,
+  };
+
+  return {
+    props: {
+      value: value,
+      ...rest,
+      ref: buttonRef,
+      ...pressable.props,
+      accessibilityRole: 'radio',
+      accessibilityLabel,
+      accessibilityState: { disabled: state.disabled, selected: state.selected },
+      accessibilityActions: [{ name: 'Select', label: radioSelectActionLabel }],
+      accessibilityPositionInSet: accessibilityPositionInSet ?? info.buttonKeys.findIndex((x) => x == value) + 1,
```

Comment:
This seems a bit fragile... it depends on buttonKeys being in the same order as the Radios passed into the RadioGroup. Do we have to manually populate a11yPosInSet (i.e. Windows won't do the work for us?) If so we might need to figure out a more robust way to keep track of child ordering

Reply:
The way buttonKeys is populated in RadioGroup means that it should theoretically be in the correct order every time. I'm also not sure what scenario might occur for them _not_ to be in the same order? 

Discussed with @ksiler about this and agreed that we will keep this as is for the time being. We do have to manually populate a11yPosInSet and the alternative to this approach would be to have RadioGroup check how many children/position on render which would be more expensive as it would re-count on every re-render.

Reply:
Ah I misunderstood how the buttonKeys info was populated. I thought it was a prop to radioGroup for some reason.